libp2p/messagesHandler.go

```diff
@@ -0,0 +1,668 @@
+package libp2p
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-p2p"
+	"github.com/ElrondNetwork/elrond-go-p2p/data"
+	"github.com/libp2p/go-libp2p-pubsub"
+	pubsubPb "github.com/libp2p/go-libp2p-pubsub/pb"
+	libp2pCrypto "github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+// TODO remove the header size of the message when commit d3c5ecd3a3e884206129d9f2a9a4ddfd5e7c8951 from
+// https://github.com/libp2p/go-libp2p-pubsub/pull/189/commits will be part of a new release
+var messageHeader = 64 * 1024 // 64kB
+var maxSendBuffSize = (1 << 21) - messageHeader
+
+const durationBetweenSends = time.Microsecond * 10
+
+// ArgMessagesHandler is the DTO struct used to create a new instance of messages handler
+type ArgMessagesHandler struct {
+	PubSub             PubSub
+	DirectSender       p2p.DirectSender
+	Throttler          core.Throttler
+	OutgoingCLB        ChannelLoadBalancer
+	TopicsHandler      TopicsHandler
+	Marshaller         p2p.Marshaller
+	ConnMonitorWrapper p2p.ConnectionMonitorWrapper
+	PeersRatingHandler p2p.PeersRatingHandler
+	Debugger           p2p.Debugger
+	SyncTimer          p2p.SyncTimer
+	IDProvider         IDProvider
+}
+
+// TODO[Sorin]: add unit tests
+type messagesHandler struct {
+	ctx                context.Context
+	cancelFunc         context.CancelFunc
+	pb                 PubSub
+	ds                 p2p.DirectSender
+	throttler          core.Throttler
+	outgoingCLB        ChannelLoadBalancer
+	topicsHandler      TopicsHandler
+	marshaller         p2p.Marshaller
+	connMonitorWrapper p2p.ConnectionMonitorWrapper
+	peersRatingHandler p2p.PeersRatingHandler
+	debugger           p2p.Debugger
+	syncTimer          p2p.SyncTimer
+	idProvider         IDProvider
```

Comment:
since the ID does not change during the lifetime of the messagesHandler, you could have passed it as a core.PeerID value

Reply:
updated and removed IDProvider interface